Update docs for Project Templates Public Preview (#3185)
OctopusDeploy/docs: src/pages/docs/platform-hub/templates/project-templates/index.md

---

:::div{.warning}

> Project templates are in Public Preview. The feature is still evolving and standard SLAs don't apply. We don't recommend it for production workloads yet. It's available to Enterprise customers on Cloud and to self-hosted customers running Octopus 2026.2. We'd love your feedback as we work towards general availability.

Reviewer: Title case for "general availability": "General Availability"?

Author: Fixed!

> Project templates are in Public Preview. The feature is still evolving and standard SLAs don't apply. We don't recommend it for production workloads yet. It's available to Enterprise customers on Cloud and to self-hosted customers running Octopus 2026.2. We'd love your feedback as we work towards general availability.

Reviewer 2: ```suggestion
Project templates are in Public Preview. The feature is still evolving and standard SLAs don't apply. We don't recommend it for production workloads yet. It's available to Enterprise customers on Cloud and to self-hosted customers running Octopus 2026.2. We'd love your feedback as we work towards general availability, [tell us what you think.](https://roadmap.octopus.com/c/263-project-templates).
```

Author: Fixed!

:::

## Overview

Project templates are reusable project blueprints that can be shared across multiple spaces in Octopus Deploy. Instead of manually configuring each new project from scratch, defining deployment steps and variables every time, you create a single template that any space can use as a starting point. This ensures teams follow the same standards and removes the risk of configuration drift.

> To create or manage your project templates, navigate to Platform Hub. If you haven't set up your Git repository, you must do so before creating a project template.

Reviewer: Link to this section?

https://octopus.com/docs/platform-hub#github-app-connections-in-platform-hub

Author: Fixed!

:::

:::div{.hint}
Unlike standard projects, project templates validate the deployment process when you publish, not when you commit. You can save an incomplete process and continue configuring parameters and variables before publishing. This will change once we add inline variable configuration to the deployment process editor.
:::

:::div{.hint}
If your deployment process includes a process template configured to auto-update on patch or minor versions, those updates flow through to templated projects automatically, even without you publishing a new version of the project template. This means two releases created on different days could use different versions of the process template, without anyone making any change to the project template or the project itself.

The project template process editor shows a warning callout when included process templates have received automatic updates since the template was last published. Use it to review the changes before publishing a new version.

> We're interested in your [feedback](#feedback) on whether this behavior meets your expectations.

Reviewer 3: Nice addition 👍

> We're interested in your [feedback](#feedback) on whether this behavior meets your expectations.

Reviewer 2: Let's not use the feedback button at the bottom of the page as it's specific to the docs and not the feature.

```suggestion
We're interested in your [feedback](https://roadmap.octopus.com/c/263-project-templates) on whether this behavior meets your expectations.
```

Author: Fixed!

:::

## Parameters

Parameters let you define the inputs a user must supply when they create a project from the template. They're the mechanism for making a template flexible. Rather than hardcoding values that differ between teams or spaces, you expose them as parameters.

:::div{.warning}

> Project templates don't yet support parameter scoping or sensitive parameter values. We're still shaping how parameters, variables, and scoping work together and expect this area to evolve. We'd love your [feedback](#feedback).

Reviewer 3: We do support sensitive parameter _consumer_ values, but not sensitive default values. It might be worth making this a little clearer
```suggestion
Project templates don't yet support parameter scoping or sensitive parameter default values. We're still shaping how parameters, variables, and scoping work together and expect this area to evolve. We'd love your [feedback](#feedback).
```

Author: Fixed!

Reviewer 2: Again, let's use the same feedback URL:
- https://roadmap.octopus.com/c/263-project-templates

Variable values can reference parameters, letting you combine fixed template-level values with project-supplied inputs where needed.

### Variable types

Project template variables support the following types:

- **Text**: plain string values.
- **Sensitive**: encrypted values like passwords and API keys.
- **Account**: references to accounts defined in Platform Hub. Add the account in Platform Hub before using it in a template.
- **Certificate**: references to certificates defined in Platform Hub.

### Variable scoping

You can scope a project template variable to any combination of the following:

- A specific step in the deployment process.
- A process template usage, when the template's deployment process includes one.

> - An environment parameter, target tag parameter, or tenant tag parameter defined on the template. The variable applies wherever the project's parameter value resolves to.

Reviewer 3: You can scope to multiple of each of these. What do you think about something along these lines? (includes minor rewording of last sentence)

```suggestion
- Specific steps in the deployment process.
- Process template usages, when the template's deployment process includes one or more.
- Environment parameters, target tag parameters, or tenant tag parameters defined on the template. Whatever the project's parameter value resolves to determines the scope.
```

Author: Fixed!

## Feedback

> Project templates are in Public Preview and we're actively shaping how the feature works. If you run into something unexpected or have thoughts on how parameters, variables, scoping, or anything else should work, we'd love to hear from you. [Share your feedback](https://oc.to/feedback) to help us build this the right way.

Reviewer 2: ```suggestion
Project templates are in Public Preview and we're actively shaping how the feature works. If you run into something unexpected or have thoughts on how parameters, variables, scoping, or anything else should work, we'd love to hear from you. [Share your feedback](https://roadmap.octopus.com/c/263-project-templates) to help us build this the right way.
```
